cdk-infra/utils/buckets.ts

```diff
@@ -10,3 +16,149 @@ export const docsBucketNames = [
   'docs-java',
   'docs-node',
 ] as const;
+
+interface CustomBucketProps {
+  scope: Construct;
+  featureName: string;
+  env: AutoBuilderEnv;
+  bucketName: string;
+}
+export function createCustomBucket({ scope, featureName, env, bucketName }: CustomBucketProps): Bucket {
```

Comment:
This logic existed previously in the `bucket-construct.ts` file, but moved it here since it isn't always needed. It's also nice to separate this out into its own function in a separate file.